basxconnect/core/layouts/editperson.py

```diff
@@ -517,6 +516,58 @@ def grid_inside_tab(*elems, **attrs):
     return layout.grid.Grid(*elems, **attrs)
 
 
+def tile_with_edit_modal(modal_view):
+    modal = layout.modal.Modal.with_ajax_content(
+        heading=f"Edit {modal_view.heading()}",
+        url=hg.F(
+            lambda c, e: reverse(
+                modal_view.path(),
+                kwargs={"pk": c["object"].pk},
+                query={"asajax": True},
+            )
+        ),
+        submitlabel="save",
+    )
+    displayed_fields = [display_field_value(field) for field in modal_view.fields()]
+    return tiling_col(
+        R(C(hg.H4(_(modal_view.heading())))),
+        *displayed_fields,
+        R(
+            C(
+                layout.button.Button(
+                    "Edit", buttontype="tertiary", icon="edit", **modal.openerattributes
+                ),
+                modal,
+            ),
+        ),
+        width=8,
+    )
+
+
+def display_field_value(field):
```

Comment:
This kind of functionality is required quite often. There exists ```bread.layout.base.fieldlabel```, but it does only work with concrete models (so you have to embed it inside ```hg.F``` in order to get the model form ```c["object"]```. It might be worth to add a class or a function to bread.layout.base which returns a Lazy object which renders the according field label.
Something similar exists for object values at ```bread.layout.base. FormattedContextValue```. It could be used in your case. However, the ```format_value``` does a little bit too much magic since it tried to render anything nice, like related fields as lists.
It might not be necessary to work on that right now, but something similar to the two hg.F elements we have in this function are needed in many places, so having something unified will be usefull.